pkgs/development/coq-modules/wasmcert/default.nix

```diff
@@ -0,0 +1,81 @@
+{
+  lib,
+  mkCoqDerivation,
+  callPackage,
+  which,
+  coq,
+  flocq,
+  parseque,
+  mathcomp-ssreflect,
+  compcert,
+  ExtLib,
+  makeWrapper,
+  version ? null,
+}:
+
+with lib;
+mkCoqDerivation {
+  pname = "wasm";
+  repo = "WasmCert-Coq";
+  owner = "WasmCert";
+
+  inherit version;
+  defaultVersion =
+    let
+      case = coq: mc: out: {
+        cases = [
+          coq
+          mc
+        ];
+        inherit out;
+      };
+    in
+    with lib.versions;
+    lib.switch
+      [ coq.coq-version mathcomp-ssreflect.version ]
+      [
+        (case (isEq "8.20") (isEq "2.4") "2.1.0")
+      ]
+      null;
+
+  release."2.1.0".sha256 = "sha256-k094mxDLLeelYP+ABm+dm6Y5YrachrbhNeZhfwLHNRo=";
+
+  mlPlugin = true;
+  useDune = true;
+
+  propagatedBuildInputs = [
+    ExtLib
+    mathcomp-ssreflect
+    parseque
+    flocq
+    compcert
+  ];
+
+  buildInputs = [
+    coq.ocamlPackages.mdx
+    coq.ocamlPackages.linenoise
+    coq.ocamlPackages.wasm
+    makeWrapper
+  ];
+
+  releaseRev = v: "v${v}";
+
+  installPhase = ''
+    dune install -p coq-wasm --prefix=$out --libdir $OCAMLFIND_DESTDIR
+    wrapProgram $out/bin/wasm_coq_interpreter --prefix OCAMLPATH : $OCAMLPATH
+
+    # Dune places the Coq files in an unexpected folder structure, the following fixes that
```

Comment:
Without the following lines, one can't import the package.
I think this is because dune places the *.vo files in `$out/lib/ocaml/4.14.2/site-lib/coq/user-contrib/Wasm`, instead of the required `$out/lib/coq/8.20/user-contrib/Wasm`. 

Is that right/is there a better way of doing it?

Reply:
Maybe using the `preInstall` and/or `postInstall` hook can save some ducplication from what mkCoqDerivation already does: https://github.com/NixOS/nixpkgs/blob/6cb4a27afaff139a7a831363057108d632e265f2/pkgs/build-support/coq/default.nix#L237-L241